src/sentry/shared_integrations/client/base.py

```diff
@@ -143,19 +144,20 @@ def _request(
                 self.track_response_data("timeout", span, e)
                 raise ApiTimeoutError.from_exception(e) from e
             except HTTPError as e:
-                resp = e.response
-                if resp is None:
-                    self.track_response_data("unknown", span, e)
-
-                    # It shouldn't be possible for integration_type to be null.
-                    extra = {"url": full_url}
-                    if self.integration_type:
-                        extra[self.integration_type] = self.name
-                    self.logger.exception("request.error", extra=extra)
-
-                    raise ApiError("Internal Error", url=full_url) from e
-                self.track_response_data(resp.status_code, span, e)
-                raise ApiError.from_response(resp, url=full_url) from e
+                self._handle_HTTPError(e, full_url, span)
```

Comment:
Thanks for making the code cleaner ❤️

Reply:
Ah, well, the best laid plans... Yes, `InvalidChunkLength` _is_ an `HTTPError`. But it's an `HTTPError` from `urllib3`, whereas the `HTTPError`s we were already dealing with are from `requests`, and it turns out they're not compatible. So... so much for factoring that code out into a function.